### PR TITLE
Generate SDK using a generator 

### LINKED ISSUE #1: Generate SDK using a generator

### PR DESCRIPTION
**Pull request by Aditya-Kumar-Code**
Will Close #1

### CODE REVIEW

File: .DS_Store
Comment (sutyum) [inline .DS_Store:1]: delete and gitignore

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+from setuptools import setup, find_packages  # noqa: H301
+
+# To install the library, run the following
+#
+# python setup.py install
+#
+# prerequisite: setuptools
+# http://pypi.python.org/pypi/setuptools
+NAME = "openapi-client"
```
Comment (sutyum) [inline setup.py]: regenerate with an appropriate name: `abdm-hiu-sdk`

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "urllib3 >= 1.25.3, < 2.1.0",
+    "python-dateutil",
+    "pydantic >= 2",
+    "typing-extensions >= 4.7.1",
+]
+
+setup(
+    name=NAME,
+    version=VERSION,
+    description="FastAPI",
```
Comment (sutyum) [inline setup.py]: use an apt description: "Python SDK for ABDM HIU REST API"

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    author="OpenAPI Generator community",
+    author_email="team@openapitools.org",
+    url="",
+    keywords=["OpenAPI", "OpenAPI-Generator", "FastAPI"],
```
Comment (sutyum) [inline setup.py]: `keywords=["ABDM", "SDK", "UHI"],`

**Review by sutyum — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,71 @@
+[tool.poetry]
+name = "openapi_client"
```
Comment (sutyum) [inline pyproject.toml]: seems you are executing the cli without setting all the important paramaters. for instance `package-name` etc

**Review by sutyum — CHANGES_REQUESTED** [review]

File: openapi_client/__init__.py
Comment (sutyum) [inline openapi_client/__init__.py:1]: Use the CLI correctly, make sure to specify various variables in the CLI params in order to control the generated output. Ideally the generated SDK should have correct package name (abdm-hiu-sdk), description, etc.

The more carefully you fill the CLI params, the higher the quality of output you should get.
```sh
NAME
        openapi-generator-cli generate - Generate code with the specified
        generator.

SYNOPSIS
        openapi-generator-cli generate
                [(-a <authorization> | --auth <authorization>)]
                [--api-name-suffix <api name suffix>] [--api-package <api package>]
                [--artifact-id <artifact id>] [--artifact-version <artifact version>]
                [(-c <configuration file> | --config <configuration file>)] [--dry-run]
                [(-e <templating engine> | --engine <templating engine>)]
                [--enable-post-process-file]
                [(-g <generator name> | --generator-name <generator name>)]
                [--generate-alias-as-model] [--git-host <git host>]
                [--git-repo-id <git repo id>] [--git-user-id <git user id>]
                [--global-property <global properties>...] [--group-id <group id>]
                [--http-user-agent <http user agent>]
                [(-i <spec file> | --input-spec <spec file>)]
                [--ignore-file-override <ignore file override location>]
                [--import-mappings <import mappings>...]
                [--instantiation-types <instantiation types>...]
                [--invoker-package <invoker package>]
                [--language-specific-primitives <language specific primitives>...]
                [--legacy-discriminator-behavior] [--library <library>]
                [--log-to-stderr] [--minimal-update]
                [--model-name-prefix <model name prefix>]
                [--model-name-suffix <model name suffix>]
                [--model-package <model package>]
                [(-o <output directory> | --output <output directory>)] [(-p <additional properties> | --additional-properties <additional properties>)...]
                [--package-name <package name>] [--release-note <release note>]
                [--remove-operation-id-prefix]
                [--reserved-words-mappings <reserved word mappings>...]
                [(-s | --skip-overwrite)] [--server-variables <server variables>...]
                [--skip-validate-spec] [--strict-spec <true/false strict behavior>]
                [(-t <template directory> | --template-dir <template directory>)]
                [--type-mappings <type mappings>...] [(-v | --verbose)]

OPTIONS
        -a <authorization>, --auth <authorization>
            adds authorization headers when fetching the OpenAPI definitions
            remotely. Pass in a URL-encoded string of name:header with a comma
            separating multiple values

...... (results omitted)

        -v, --verbose
            verbose mode
```

**Review by sutyum — CHANGES_REQUESTED** [review]

File: .DS_Store
Comment (Aditya-Kumar-Code) [reply .DS_Store:1]: Done

File: setup.py (hunk 1 above)
Comment (Aditya-Kumar-Code) [reply setup.py]: Done

File: setup.py (hunk 2 above)
Comment (Aditya-Kumar-Code) [reply setup.py]: Done

File: setup.py (hunk 3 above)
Comment (Aditya-Kumar-Code) [reply setup.py]: Done

File: pyproject.toml (hunk 4 above)
Comment (Aditya-Kumar-Code) [reply pyproject.toml]: Done

File: openapi_client/__init__.py
Comment (Aditya-Kumar-Code) [reply openapi_client/__init__.py:1]: Done

File: examples/consent_change_notification.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import abdm_hiu_sdk
+from abdm_hiu_sdk.rest import ApiException
```
Comment (sutyum) [inline examples/consent_change_notification.py]:
```py
import abdm
from abdm import ApiException
```

File: abdm-hiu-sdk/__init__.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+"""  # noqa: E501
+
+
+__version__ = "1.0.0"
+
+# import apis into sdk package
+from abdm-hiu-sdk.api.default_api import DefaultAPI
+
+# import ApiClient
+from abdm-hiu-sdk.api_response import ApiResponse
```
Comment (sutyum) [inline abdm-hiu-sdk/__init__.py]: `abdm-hiu-sdk` -> `abdm_hiu_sdk` -> `abdm`

**Review by sutyum — CHANGES_REQUESTED** [review]

**Review by sutyum — APPROVED** [review]

File: .github/workflows/abdm.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
+# NOTE: This file is auto generated by OpenAPI Generator.
+# URL: https://openapi-generator.tech
+#
+# ref: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: abdm Python package
+
+on: [push, pull_request]
```
Comment (sutyum) [inline .github/workflows/abdm.yml]: on pull_request only

**Review by sutyum — CHANGES_REQUESTED** [review]

**Comment by sutyum** [comment]
Pass the tests in order to merge

File: examples/consent_change_notification.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import abdm
+from abdm.rest import ApiException
```
Comment (sutyum) [inline examples/consent_change_notification.py:2]: `from abdm.rest import ApiException` these should have been `from abdm import ApiException`

File: hiu_api.py
Comment (sutyum) [inline hiu_api.py:1]: why is this file needed?

**Review by sutyum — CHANGES_REQUESTED** [review]

File: LICENSE
Comment (sutyum) [inline LICENSE:1]: no need to delete this LICENSE

**Review by sutyum — CHANGES_REQUESTED** [review]

File: .DS_Store
Comment (sutyum) [inline .DS_Store:1]: should have been gitignored

File: .gitignore
Comment (sutyum) [inline .gitignore:1]: shouldn't have been deleted

**Review by sutyum — CHANGES_REQUESTED** [review]
gitignored files have been committed and gitignore file deleted, etc

**Review by sutyum — CHANGES_REQUESTED** [review]
all the __pycache__ dirs need to be deleted, along with .DS_Store

**Comment by sutyum** [comment]
your workflow is currently running type checking on the test dir as well. You can skip typechecking in the test dir

File: LICENSE [hunk 9]
```diff
@@ -18,4 +18,4 @@ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-SOFTWARE.
+SOFTWARE.
```
Comment (sutyum) [inline LICENSE:21]: revert back to original

**Review by sutyum — CHANGES_REQUESTED** [review]

**Review by sutyum — CHANGES_REQUESTED** [review]
As mentioned earlier, all the __pycache__ folders need to be deleted

File: .github/workflows/abdm.yml [hunk 10]
```diff
@@ -0,0 +1,78 @@
+# NOTE: This file is auto generated by OpenAPI Generator.
+# URL: https://openapi-generator.tech
+#
+# ref: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: abdm Python package
```
Comment (sutyum) [inline .github/workflows/abdm.yml]: rename to "Check and Tests"

**Review by sutyum — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 9 above)
Comment (sutyum) [reply LICENSE:21]: still pending

**Review by sutyum — CHANGES_REQUESTED** [review]

File: LICENSE (hunk 9 above)
Comment (Aditya-Kumar-Code) [reply LICENSE:21]: sir not able to understood revert back to original

has used the same license as it is in dev branch
